hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java

```diff
@@ -73,7 +73,7 @@
  * When there is a {@link RocksDBException} with error,
  * this class will close the underlying {@link org.rocksdb.RocksObject}s.
  */
-public final class RocksDatabase {
+public final class RocksDatabase implements Closeable {
```

Comment:
Can we add the close in the finalize method of this object as a fail safe, & log a warning if it is not being closed properly with the stacktrace of creation of the object. Would this of any help?

Reply:
It is not advised to override the finalized method to close object. For more check: https://softwareengineering.stackexchange.com/questions/288715/is-overriding-object-finalize-really-bad/288724#288724

I added a WARN log statement instead of this to catch any memory leak.